src/compas_timber/_fabrication/lap.py

```diff
@@ -0,0 +1,511 @@
+        # calculate the length of the lap based on the intersection points of the planes with the reference edge
+        # the length is the perpendicular distance between the two planes
+        vect = ref_side.yaxis.cross(planes[0].normal)
+        intersection_pts = [Point(*intersection_line_plane(ref_edge, plane)) for plane in planes]
+        dist = distance_point_point(*intersection_pts)
+        # calculate the compansation distance based on the angle of intersection
+        compensation_angle = angle_vectors_signed(vect, ref_side.xaxis, ref_side.normal)
+        dist += depth / abs(math.tan(compensation_angle))
+        # calculate the perpendicular distance between the two planes
+        length = math.sin(math.radians(angle))*dist
+        return length
+
+
+    ########################################################################
+    # Methods
+    ########################################################################
+
+    def apply(self, geometry, beam):
+        """Apply the feature to the beam geometry.
+
+        Parameters
+        ----------
+        geometry : :class:`~compas.geometry.Brep`
+            The beam geometry to be cut.
+        beam : :class:`compas_timber.elements.Beam`
+            The beam that is cut by this instance.
+
+        Raises
+        ------
+        :class:`~compas_timber.elements.FeatureApplicationError`
+            If the cutting plane does not intersect with beam geometry.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Brep`
+            The resulting geometry after processing
+
+        """
+        # type: (Brep, Beam) -> Brep
+        box = self.volume_from_params_and_beam(beam)
+        lap_volume = Brep.from_box(box)
+        try:
+            return geometry - lap_volume
+        except BrepError:
+            raise FeatureApplicationError(
+                lap_volume,
+                geometry,
+                "The lap volume does not intersect with the beam geometry.",
+            )
+
+    def volume_from_params_and_beam(self, beam):
+        """Calculates the volume of the cut from the machining parameters in this instance and the given beam
+
+        Parameters
+        ----------
+        beam : :class:`compas_timber.elements.Beam`
+            The beam that is cut by this instance.
+
+        Returns
+        -------
+        :class:`compas.geometry.Box`
+            The boxvolume of the cut as a box.
+
+        """
+        # type: (Beam) -> Brep
+
+        ref_side = beam.side_as_surface(self.ref_side_index)
+        p_origin = ref_side.point_at(self.start_x, self.start_y)
+
+        box_frame = Frame(p_origin, ref_side.frame.xaxis, -ref_side.frame.yaxis)
+        rot_angle = 90-self.angle if self.orientation == OrientationType.END else self.angle-90
+        box_frame.rotate(math.radians(rot_angle), box_frame.normal, point=box_frame.point)
+
+        box = Box(xsize=self.length, ysize=self.width, zsize=self.depth, frame=box_frame)
```

Comment:
discussed with @papachap, I agree we should at least be able to do what was possible before AKA arbitrary laps as a result of lap joinery.

a potential solution would be a `from_beam_beam()` constructor that follows similar logic as the current lap joint. see implementation by @jonashaldemann: https://github.com/gramaziokohler/compas_timber/blob/295c3ffad6a55ebf69fd55a691948e994dfceff2/src/compas_timber/connections/lap_joint.py#L128

Reply:
This is true that it is a simplification. For sure It is wise to implement an additional alternative constructor as @chenkasirer suggested in order to allow for non box-like laps by using the two beams to extract the params.

Also the geometrical representation can be done with a polyhedron volume rather than a box.

In regards to the joint definitions of TButtJoint & LButtJoint though, this simplification existed from before as the inclination was set to 90 degrees by default. See: https://github.com/gramaziokohler/compas_timber/blob/295c3ffad6a55ebf69fd55a691948e994dfceff2/src/compas_timber/connections/t_butt.py#L86
But it doesn't require the beams to be orthogonal, it will just create a lap as the one you can see in the picture below.

![image](https://github.com/user-attachments/assets/0e377fc0-e6d4-4b88-abab-6fb38334c8ed)


The `_create_negative_volumes()` was used only for the `XHalfLapJoint`. 

Let me know if we want to keep this simplification to any of those joint definitions.